src/bin/psql/tab-complete.c

```diff
@@ -4244,6 +4260,21 @@ get_previous_words(int point, char **previous_words, int nwords)
 	}
 }
 
+/*
+ * Returns true if the given word ends with a right parenthesis ')'.
+ * Returns false if the input is NULL or the word is empty.
+ * This function is typically used to check if a SQL token ends with ')',
+ * for context-sensitive tab completion.
+ */
+static bool
+ends_with_paren(const char *word)
```

Comment:
Do we really need this function? It seems that it is impossible that word is NULL or empty string

Reply:
This function provides a more readable alternative to checking `prev_wd[len - 1] == ')'`. Since this check is performed in several places, I decided to move it into a separate function. The NULL and non-empty string checks are necessary to safely evaluate the expression `word[len - 1] == ')'`.